sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/test/scala/com/azure/cosmos/spark/CosmosClientConfigurationSpec.scala

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.spark
+
+import com.azure.cosmos.ConsistencyLevel
+
+class CosmosClientConfigurationSpec extends UnitSpec {
+    //scalastyle:off multiple.string.literals
+
+    "CosmosClientConfiguration" should "parse configuration" in {
+        val userConfig = Map(
+            "spark.cosmos.accountEndpoint" -> "https://localhsot:8081",
+            "spark.cosmos.accountKey" -> "xyz",
+            "spark.cosmos.accountConsistency" -> "Strong"
+        )
+
+        val configuration = CosmosClientConfiguration(userConfig)
+
+        configuration.endpoint shouldEqual userConfig("spark.cosmos.accountEndpoint")
+        configuration.key shouldEqual userConfig("spark.cosmos.accountKey")
+        configuration.consistencyLevel shouldEqual ConsistencyLevel.STRONG
+    }
+
+    it should "use default for consistency" in {
+        val userConfig = Map(
+            "spark.cosmos.accountEndpoint" -> "https://localhsot:8081",
+            "spark.cosmos.accountKey" -> "xyz"
+        )
+
+        val configuration = CosmosClientConfiguration(userConfig)
+
+        configuration.endpoint shouldEqual userConfig("spark.cosmos.accountEndpoint")
+        configuration.key shouldEqual userConfig("spark.cosmos.accountKey")
+        configuration.consistencyLevel shouldEqual ConsistencyLevel.EVENTUAL
+    }
+
+    it should "use default for consistency if wrong value" in {
```

Comment:
we should actually throw error rather than using the default. if there is anything wrong we should error to avoid customer unintentional spelling issues in configuration.